src/librustc_typeck/diagnostics.rs

```diff
@@ -4095,6 +4095,91 @@ assert_eq!(!Question::No, true);
 ```
 "##,
 
+E0604: r##"
+A cast to `char` was attempted on another type than `u8`.
```

Comment:
I think it comes straight from the C language. In C, char are 8 bits and supposed to represent ascii characters.